pkg/scheduler/scheduler.go

```diff
@@ -114,3 +123,83 @@ type VPod interface {
 
 	GetResourceVersion() string
 }
+
+type ScaleCache struct {
+	lock                 sync.RWMutex //protects access to entries, entries itself is concurrency safe, so we only need to ensure that we correctly access the pointer
+	entries              *cache.Expiring
+	statefulSetClient    clientappsv1.StatefulSetInterface
+	statefulSetNamespace string
```

Comment:
It seems like the scheduler package is agnostic from statefulset, so it seems this scale cache needs to be in a statefulset specific package

Reply:
Or we could redesign this Cache to be statefulset agnostic with a custom interface

```
type Client interface {
   GetScale(ctx context.Context, name string, options metav1.GetOptions) *autoscalingv1.Scale

   UpdateScale(ctx context.Context, name string, scale *autoscalingv1.Scale, options metav1.UpdateOptions) *autoscalingv1.Scale
}

Reply:
Yeah, I think the interface will work better- I originally had it in the statefulset package but then that caused cyclic imports into the state package.

Let me try with the interface